plugin/type_uuid/item_uuidfunc.cc

```diff
@@ -29,3 +29,32 @@ String *Item_func_sys_guid::val_str(String *str)
   my_uuid2str(buf, const_cast<char*>(str->ptr()), 0);
   return str;
 }
+
+
+bool Item_func_uuid_timestamp::fix_length_and_dec(THD *thd)
+{
+  Type_std_attributes::set(
+    Type_temporal_attributes_not_fixed_dec(MAX_DATETIME_WIDTH,
+                                           TIME_SECOND_PART_DIGITS, false),
+    DTCollation_numeric());
+  set_maybe_null();
+  return FALSE;
+}
+
+
+bool Item_func_uuid_timestamp::val_native(THD *thd, Native *to)
+{
+  Type_handler_uuid_new::Fbt_null uuid(args[0]);
+  if (uuid.is_null())
+    return (null_value= true);
```

Comment:
Every `return`  contains this `null_value` assignment. This means it makes sense to extract a function:

Make a private funciton which does all the conversion. 
In this public to_native, call that function and assign null_value.